dialogue-httpurlconnection-client/src/main/java/com/palantir/dialogue/httpurlconnection/HttpUrlConnectionChannels.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.dialogue.httpurlconnection;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.conjure.java.client.config.ClientConfiguration;
+import com.palantir.dialogue.Channel;
+import com.palantir.dialogue.blocking.BlockingChannelAdapter;
+import com.palantir.dialogue.core.Channels;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/** Simple Channel factory which produces channels based on HttpURLConnection. */
+public final class HttpUrlConnectionChannels {
+
+    private HttpUrlConnectionChannels() {}
+
+    public static Channel create(ClientConfiguration conf, UserAgent baseAgent, TaggedMetricRegistry metrics) {
```

Comment:
Low importance as I think this will be used for benchmarking over prod usage, but might be worth doing the same ClientConf preconditions here?

Reply:
Perhaps in a follow-up, I have a WIP to move some of the validation to Channels.create we pick the node selection strategy, qos, etc.

I'll open a PR to update the readme with a description of each of our current clients and describe our confidence (e.g. don't use java9 or httpurlconnection)